proof-producer/bin/proof-producer/CMakeLists.txt

```diff
@@ -80,6 +80,10 @@ function(setup_proof_generator_target)
         target_compile_options(${ARG_TARGET_NAME} PRIVATE "-fconstexpr-ops-limit=4294967295")
     endif ()
 
+    if(PROOF_PRODUCER_STATIC_BINARIES)
+        target_link_options(${ARG_TARGET_NAME} PRIVATE -static -static-libgcc -static-libstdc++)
```

Comment:
Is there any reason for using `target_link_options` instead of `target_link_libraries`?

Reply:
Is there an option in `target_link_libraries` to force a linker use `-static` flag?

Reply:
https://cmake.org/cmake/help/latest/command/target_link_libraries.html
```
Added in version 3.13: [LINK_OPTIONS](https://cmake.org/cmake/help/latest/prop_tgt/LINK_OPTIONS.html#prop_tgt:LINK_OPTIONS) target property and [target_link_options()](https://cmake.org/cmake/help/latest/command/target_link_options.html#command:target_link_options) command. For earlier versions of CMake, use [LINK_FLAGS](https://cmake.org/cmake/help/latest/prop_tgt/LINK_FLAGS.html#prop_tgt:LINK_FLAGS) property instead.
```

I think `set_target_properties(${ARG_TARGET_NAME} LINK_OPTIONS ...)` + `target_link_libraries` is a bit more transparent approach

Reply:
AFAIS `target_link_libraries` doesn't allow me to set options conditionally (depending on `PROOF_PRODUCER_STATIC_BINARIES`). And using temporary variable or `set_target_properties` seems less clear to me then `target_link_options` that is intended for this usecase.

Reply:
target link options documentation actually is not clear about adding libraries though this call (it just says `any options can be passed, but there target_link_libraries call which is precisely for libraries`)

I do not think it is a big deal since linkage is not so complicated here, but cmake tends to provide some really unexpected side effects in such cases, so I am a bit concerned :)

anyways, up to you

Reply:
Unfortunately cmake doesn't have clear documentation for `target_link_libraries` either :( Seems it also accepts both link flags and libraries in LINK_OPTIONS section. So no differences to me.